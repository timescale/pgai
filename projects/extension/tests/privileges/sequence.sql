\set users {bob,fred,alice,jill}

-- check sequence privileges
\! rm -f sequence.actual
select
  n.nspname as "schema"
, k.relname as "table"
, u as "user"
, p as "privilege"
, case has_table_privilege(u, k.oid, p) when true then 'YES' else 'no' end as granted
from unnest(:'users'::text[]) u
inner join pg_namespace n on (n.nspname = any(array['ai', 'wiki']))
inner join pg_class k on (n.oid = k.relnamespace and k.relkind in ('S'))
cross join unnest(array['select', 'update']) p
order by n.nspname, k.relname, u, p
\g (format=aligned) sequence.actual
