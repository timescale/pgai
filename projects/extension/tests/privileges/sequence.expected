 schema |       table       | user  | privilege | granted 
--------+-------------------+-------+-----------+---------
 ai     | vectorizer_id_seq | alice | select    | YES
 ai     | vectorizer_id_seq | alice | update    | YES
 ai     | vectorizer_id_seq | bob   | select    | no
 ai     | vectorizer_id_seq | bob   | update    | no
 ai     | vectorizer_id_seq | fred  | select    | no
 ai     | vectorizer_id_seq | fred  | update    | no
 ai     | vectorizer_id_seq | jill  | select    | YES
 ai     | vectorizer_id_seq | jill  | update    | YES
 wiki   | post_id_seq       | alice | select    | YES
 wiki   | post_id_seq       | alice | update    | YES
 wiki   | post_id_seq       | bob   | select    | no
 wiki   | post_id_seq       | bob   | update    | no
 wiki   | post_id_seq       | fred  | select    | no
 wiki   | post_id_seq       | fred  | update    | no
 wiki   | post_id_seq       | jill  | select    | no
 wiki   | post_id_seq       | jill  | update    | no
(16 rows)

